src/python_framework_v02/troute/nhd_io.py

```diff
@@ -1084,3 +1086,75 @@ def build_coastal_ncdf_dataframe(coastal_ncdf):
     with xr.open_dataset(coastal_ncdf) as ds:
         coastal_ncdf_df = ds[["elev", "depth"]]
         return coastal_ncdf_df.to_dataframe()
+
+def lastobs_df_output(main_start_time,
+    dt,
+    nts,
+    run_results,
+    wrf_hydro_lastobs_file,
+    q0,
+    gages,
+    run_set_iterator,
+    lastobs_output_title,
+    lastobs_output_folder=False,
+    ):
+    lastobs_df = new_lastobs(run_results, dt * nts)
+    lastobs_df.index.names = ['link']
+    lastobs_df.insert(0, 'last_model_discharge', q0.loc[lastobs_df.index]['qu0'])
+    lastobs_df.insert(0, 'gages', gages)
+    if lastobs_output_folder:
+        lastobs_string = lastobs_output_folder+lastobs_output_title
+        # lastobs_string = lastobs_output_folder+"lastobs_df_"+str(run_set_iterator)+".nc"
+    else:
+        lastobs_string = lastobs_output_title
+        # lastobs_string = "lastobs_df_"+str(run_set_iterator)+".nc"
+    if not 'modelTimeAtOutput' in lastobs_df.columns:
+        lastobs_df.insert(loc=0, column='modelTimeAtOutput', value=(time.time() - main_start_time))
+    if not 'Nudge' in lastobs_df.columns:
+        lastobs_df.insert(loc=0, column='Nudge', value=np.NaN)
+    lastobs_df['Nudge'] = np.NaN
+    lastobs_df['modelTimeAtOutput'] = (time.time() - main_start_time)
+    lastobs_df.to_xarray().to_netcdf(lastobs_string)
```

Comment:
please structure the output to be analogous to the lastobs files created by WRF hydro workflows. For example, we need the modelTimeAtOutput to be an attribute not a variable.